### PR TITLE
Add a CSP and a few headers to improve security

### PR DESCRIPTION
**Pull request by taravancil**
[Manuscript case](https://glitch.manuscript.com/f/cases/3328025/)
Remix: [~mozilla](https://mozilla.glitch.me)

## Changes
This adds a content security policy as well as a few headers to improve security and get us closer to a B Mozilla observatory score. This remix is currently [at a C+](https://observatory.mozilla.org/analyze/mozilla.glitch.me)!

## Notes
I'm a bit nervous to deploy this, because if there are sources I missed, things will break! (I'm going to keep testing the remix while this in review.) I think as soon as we deploy we should have a close eye on the [CSP reporting server](https://csp-reporting-server.glitch.me) I set up, and be prepared to ship fixes quickly. I'm especially nervous about GitHub and Facebook login.

@jennschiffer if it makes sense to you, I think it'd be smart to wait to deploy until next week, so no one has to spend the weekend monitoring the reporting server.

## Things to review
- Testing all pages and flows, checking that no scripts, styles, images or other sources are blocked
- Authentication flows...is there a way to test GitHub/Facebook login on a remix?

### CODE REVIEW

**Comment by taravancil** [comment]
Ugh, I tried to pull my merge conflict fixes on the glitch console and everything broke. Sorry, I'll try to fix it after I take a walk

**Comment by taravancil** [comment]
mozilla.glitch.me is back up 🙏🏻

**Comment by taravancil** [comment]
B+ thanks to @melissamcewen!

Screen Shot 2019-03-15 at 2 52 01 PM

**Comment by jennschiffer** [comment]
i'm getting a couple of "Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”)." errors on initial load of the page in firefox and chrome. it seems to be blocking our "how it works" video.

**Comment by taravancil** [comment]
Hmm, the "how it works" video loads for me, including the captions. You're not able to play it at all?

Oh! I wonder if it's an issue with scripts that are loaded with this:
```
function loadScript(src) {
  const script = document.createElement('script');
  script.src = src;
  script.async = true;
  document.head.appendChild(script);
}
```
Looking into it now

**Comment by jennschiffer** [comment]
It loaded in firefox after all, but i just get a black box in chrome:

Screenshot 2019-03-18 15 50 41

that "violation...vr is not allowed" error is a separate embed issue i have a pr on /glitch to fix btw

**Comment by taravancil** [comment]
Some updates:

- Our score is currently a C+, and that's the best we can do for now
- I pared down the CSP to only include policies for fonts, images, and styles - those are the ones I feel most confident about, and by having even a minimal CSP, our score is bumped from a C to a C+
- I added a redirect for `http://`

This is ready for a second round of review @jennschiffer and anyone else who's interested!

File: views/index.ejs [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <% for (let i = 0; i < styles.length; ++i) { %>
     <link rel="stylesheet" type="text/css" href="<%= styles[i] %>">
     <% } %>
-    <noscript><style>#fallback, #fallback-noscript { display: inline-block !important; } #fallback-img {display: block !important}</style></noscript>
+    <!-- <noscript><style>#fallback, #fallback-noscript { display: inline-block !important; } #fallback-img {display: block !important}</style></noscript> -->
```
Comment (clottman) [inline views/index.ejs]: @taravancil do you have any memory of why you might have done this/was it intentional to help us with CSP stuff?

**Pull request merged**